### DISCUSSION TITLE
Change asset names for better intuitive download experience of newcomers to Ferdium

### DISCUSSION THREAD

**Discussion by Alphrag** [ferdium/ferdium-app#389] [General]
Hi everyone,
It seems that the name of the assets in the release section is quite confusing and hard to get by for new users that wonder what they should download. I am suggesting to change the assets name to make them more homogeneous.
My proposition will be the following: all assets will be named with the convention 
> `Ferdium-<target_os>[-<os_option>]-<version_with_channel>[-<target_arch>].<extension>` 

where:
- `<target_os>` is one of `Mac`, `Win`, `Linux`;
- `<os_option>` is an optional string to differentiate the type of the asset such as `AutoSetup`, `Installer`, `Portable`, `Unpacked` or something similar (suggestions welcomed);
- `<version_with_channel>` is the automatic version from the builder such as `6.0.0-nightly.xx` or `6.0.0-beta.x`
- `<target_arch>` is an optional argument for specific architectures such as `arm64` for mac with arm chip, `ia32` for 32bits windows installer, or one of [`amd64`, `arm64`, `armv7l`] for the linux `deb` package;
- `<extension>` is the extension of the file (e.g. `dmg`, `zip`, `msi`, `exe`, `deb`, `AppImage`, `freebsd`, `rpm`).

So a few questions on which I would like as much feedback as possible:
* Do you think this would be better than the current naming? 
* Should the `<os_option>` and `<target_arch>` elements be merged into one? Or kept separate but next to each other instead of putting the cpu architecture at the end? 
* I assume there that linux users are aware of the package for their distro (which is also in the extension). Is it a fair assumption, or should we add another possible value in `<os_option>` to account for that?

**Comment by vraravam** [comment]
I like the proposed ordering of segments.

**Comment by SpecialAro** [comment]
I also like the proposal @Alphrag. I think this way things are better organized for end-users.

On question: won't this change mess with electron-updater utility when doing auto-updates?

**Reply by Alphrag** [reply]
I can confirm that this doesn't mess with the auto-update, since these rely on the value found in `latest.yml` which is made by the builder, and thus contain the same name as the installer itself.

**Reply by vraravam** [reply]
In that case, lets go ahead and do this change before we decide on 6.0.0

**Reply by SpecialAro** [reply]
> In that case, lets go ahead and do this change before we decide on 6.0.0

Totally agree with this! Let's implement it then

**Comment by Alphrag** [comment]
Ok, I'll add this. Last question before doing so: on windows the nsis and portable versions bundle all arch at the same time. As much as I don't really case for the nsis which will be used to install and then will be deleted by the user, I am wondering if we should have the portable versions for different arch being separate. This will provide the user with a lighter executable as shown in the image below.
![image](https://user-images.githubusercontent.com/34252790/176669587-b70b8c59-b749-416d-aed8-29dd3f21c686.png)
Is it worth having these architecture separated or not? (the one with all arch will be created in any case apparently)

**Reply by SpecialAro** [reply]
I think it is worthy if it reduces file size. Tho, are we able to not build the one created with all the arch?

**Reply by vraravam** [reply]
i dont think there's any exclusion mechanism in `electron-builder` for the target OS/architectures

**Reply by SpecialAro** [reply]
Then we keep it... no problem

**Comment by Alphrag** [comment]
Renaming has now been implemented in #407.

**Comment by vraravam** [comment]
Is there a corresponding PR in the website repo?

**Reply by Alphrag** [reply]
Not yet, I'll do that tomorrow morning, as well as making PR on the repos taking care of the different auto-update for the package managers.

**Reply by vraravam** [reply]
I can take care of the homebrew repo. If I have time, and if I release the upcoming nightly, i will review the website repo as well - if any changes are required

**Reply by vraravam** [reply]
Done changes in the https://github.com/ferdium/homebrew-ferdium repo. Could not find any changes to be done in the https://github.com/ferdium/ferdium-website repo.
